lib/main.js

```diff
@@ -121,6 +121,12 @@ function initSession(shiny, options, shinyServer) {
 
       factory(url, ctx, (err, conn) => {
         pc.resolve(err, conn);
+        conn.onclose = e => {
+          if (!options.reconnect) {
+            log("connection closed; reconnect not enabled");
+            reconnectUI.showDisconnected();
```

Comment:
Actually, @aronatkins, I think Shiny apps may use the `pc.reconnect` value to determine if it's feasible to use a Shiny-level reconnect. Joe mentioned this feature to me, although I don't understand it well.